### PR TITLE
Backed out changeset: e45723cc7580

### PR DESCRIPTION
**Pull request by zooba**

### CODE REVIEW

**Comment by zooba** [comment]
Pushed into the same branch, so now this also works around the issue we were hitting today.

File: Python/Product/PythonTools/PythonTools/Editor/AutoIndent.cs [hunk 1]
```diff
@@ -84,7 +84,10 @@ private struct LineInfo {
                 }
 
                 string sline = tokens[tokenIndex].Span.GetText();
-                var lastChar = sline.Length == 0 ? '\0' : sline[sline.Length - 1];
+                char lastChar = '\0';
```
Comment (crwilcox) [inline Python/Product/PythonTools/PythonTools/Editor/AutoIndent.cs]: What is the point to this change?

**Comment by crwilcox** [comment]
:+1:

File: Python/Product/PythonTools/PythonTools/Editor/AutoIndent.cs (hunk 1 above)
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Editor/AutoIndent.cs]: It avoids a Roslyn bug that generates invalid code and causes InvalidProgramExceptions.

**Pull request merged**